components/BottomAppBar/examples/supplemental/BottomAppBarExampleTableViewController.m

```diff
@@ -0,0 +1,52 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "BottomAppBarExampleTableViewController.h"
+
+@implementation BottomAppBarExampleTableViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  
+  self.tableView.layoutMargins = UIEdgeInsetsZero;
+  self.tableView.separatorInset = UIEdgeInsetsZero;
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return 50;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"cell"];
+  cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
```

Comment:
You're dequeueing and creating a new one. You just need to dequeue.

Reply:
Done.

Reply:
Emphasizing my previous comment: this is only true if you're using cell class registration.

The method you're using: https://developer.apple.com/documentation/uikit/uitableview/1614891-dequeuereusablecellwithidentifie
does not guarantee a returned cell. Your original code was correct.

The method Will is referring to: https://developer.apple.com/documentation/uikit/uitableview/1614878-dequeuereusablecellwithidentifie?language=objc
Note that you must call registerClass:forCellReuseIdentifier: in order for this method to work.

Reply:
Updating per Jeff's remarks.